src/components/back-link-helper/back-link-helper.js

```diff
@@ -14,10 +14,17 @@ BackLinkHelper.prototype.init = function init() {
        Currently, a page refresh sets the referer to empty, leading to the back link being hidden
        under our existing logic.
      */
-    // eslint-disable-next-line max-len
     const referrerNotOnSameDomain = () => {
       const referer = this.document.referrer;
-      return !referer || referer.indexOf(this.window.location.host) === -1;
+
+      // Allows PEGA domains
+      const allowedDomains = ['maccount-np.hmrc.gov.uk', 'account.hmrc.gov.uk'];
+      const isAllowed = allowedDomains.some((e) => referer.includes(e));
+
+      const noReferer = !referer;
+      const sameDomain = referer.indexOf(this.window.location.host) === -1;
```

Comment:
I think sameDomain here might be misnamed, I think it's checking if it's on a different domain - it looks like the changes have broken the existing tests and we've not added any new tests to cover this change

then I think rather than modifying the referrerNotOnSameDomain it might be clearer to add a new check, maybe "referrerNotOnSameDomain and referrerNotOnAllowList" type deal - because it seems we've changed the meaning of referrerNotOnSameDomain without changing it's name